mvpa2/tests/test_searchlight.py

```diff
@@ -828,6 +833,41 @@ def test_gnbsearchlight_3partitions_and_splitter(self):
         res_gnb_sl_ = gnb_sl_(ds)
         assert_datasets_equal(res_gnb_sl, res_gnb_sl_)
 
+    @sweepargs(nblocks=(1, 2, 25))
+    def test_preallocate_output(self, nblocks):
+        ds = datasets['3dsmall'].copy()[:, :25] # smaller copy
+        ds.fa['voxel_indices'] = ds.fa.myspace
+        ds.fa['feature_id'] = np.arange(ds.nfeatures)
+
+        def measure(ds):
+            # return more than one sample
+            return np.repeat(ds.fa.feature_id, 10, axis=0)
+
+        nprocs = [1, 2] if externals.exists('pprocess') else [1]
+        enable_ca = ['roi_sizes', 'raw_results', 'roi_feature_ids']
+        for nproc in nprocs:
+            sl = sphere_searchlight(measure,
+                                    radius=0,
+                                    center_ids=np.arange(ds.nfeatures),
+                                    nproc=nproc,
+                                    enable_ca=enable_ca,
+                                    nblocks=nblocks
+                                    )
+            sl_inplace = sphere_searchlight(measure,
+                                    radius=0,
+                                    preallocate_output=True,
+                                    center_ids=np.arange(ds.nfeatures),
+                                    nproc=nproc,
+                                    enable_ca=enable_ca,
+                                    nblocks=nblocks
+                                    )
+            out = sl(ds)
+            out_inplace = sl_inplace(ds)
+
+            for c in enable_ca:
+                assert_array_equal(sl.ca[c].value, sl_inplace.ca[c].value)
+        assert_array_equal(out.samples, out_inplace.samples)
+        assert_array_equal(out.fa.center_ids, out_inplace.fa.center_ids)
```

Comment:
I guess should be indented to reside under `nproc` loop? ideally, should also be checked against nprocs=1 results even for nprocs=2, right?